File: C_Sharp/AccessibilityLab/InheritanceLab/Coordinate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InheritanceLab
{
    class Coordinate
    {
        double _x;
        double _y;

        public static readonly Coordinate Origin = new Coordinate(0, 0);


        public Coordinate(double x, double y)
        {
            this._x = x;
            this._y = y;
        }

        public double X => _x;
        public double Y => _y;

        public double DistanceTo(Coordinate other)
        {
            //c^2 = a^2 + b^2

            double xDist = this._x - other._x;
            double yDist = this._y - other._y;

            double xDistSquared = xDist * xDist;
            double yDistSquared = yDist * yDist;

            double distSquared = xDistSquared + yDistSquared;
            double distance = Math.Sqrt(distSquared);
            return distance;
        }

        public double DistanceFromOrigin()
        {
            return DistanceTo(Origin);
        }

        public static double DistanceBetween(Coordinate first, Coordinate second)
        {
            //c^2 = a^2 + b^2

            double xDist = first._x - second._x;
            double yDist = first._y - second._y;

            double xDistSquared = xDist * xDist;
            double yDistSquared = yDist * yDist;

            double distSquared = xDistSquared + yDistSquared;
            double distance = Math.Sqrt(distSquared);
            return distance;
        }

    }
}

File: Java/AccessibilityLab/src/com/sonicscholar/Coordinate.java
package com.sonicscholar;

public class Coordinate {
    private int _x;
    private int _y;

    public static final Coordinate ORIGIN = new Coordinate(0,0);

    public Coordinate(int x, int y) {
        _x = x;
        _y = y;
    }

    public int getX(){ return _x;}
    public int getY(){ return _y;}

    public double distanceTo(Coordinate other){
        double xDist = this._x - other._x;
        double yDist = this._y - other._y;

        double xDistSquared = xDist * xDist;
        double yDistSquared = yDist * yDist;

        //c^2 = a^2 + b^2
        //c = sqrt(a^2 + b^2)

        double distSquared = xDistSquared + yDistSquared;
        double distance = Math.sqrt(distSquared);
        return distance;
    }

    public static double distanceBetween(Coordinate c1, Coordinate c2){
        double xDist = c1._x - c2._x;
        double yDist = c1._y - c2._y;

        double xDistSquared = xDist * xDist;
        double yDistSquared = yDist * yDist;

        //c^2 = a^2 + b^2
        //c = sqrt(a^2 + b^2)

        double distSquared = xDistSquared + yDistSquared;
        double distance = Math.sqrt(distSquared);
        return distance;
    }

    public double distanceFromOrigin(){
        return distanceTo(ORIGIN);
    }
}
